jupyter_server_fileid/manager.py

```diff
@@ -37,35 +37,174 @@ def wrapped(self, *args, **kwargs):
     return decorator
 
 
-class FileIdManager(LoggingConfigurable):
+class BaseFileIdManager(LoggingConfigurable):
     """
-    Manager that supports tracking files across their lifetime by associating
-    each with a unique file ID, which is maintained across filesystem operations.
-
-    Notes
-    -----
-
-    All private helper methods prefixed with an underscore (except `__init__()`)
-    do NOT commit their SQL statements in a transaction via `self.con.commit()`.
-    This responsibility is delegated to the public method calling them to
-    increase performance. Committing multiple SQL transactions in serial is much
-    slower than committing a single SQL transaction wrapping all SQL statements
-    performed during a method's procedure body.
+    Base class for File ID manager implementations. All File ID
+    managers should inherit from this class.
     """
 
     root_dir = Unicode(
-        help=("The root being served by Jupyter server. Must be an absolute path."), config=True
+        help=("The root directory being served by Jupyter server. Must be an absolute path."),
+        config=False,
     )
 
     db_path = Unicode(
         default_value=default_db_path,
         help=(
-            "The path of the DB file used by `FileIdManager`. "
+            "The path of the DB file used by File ID manager implementations. "
             "Defaults to `jupyter_data_dir()/file_id_manager.db`."
         ),
         config=True,
     )
 
+    @validate("root_dir", "db_path")
+    def _validate_abspath_traits(self, proposal):
+        if proposal["value"] is None:
+            raise TraitError(f"FileIdManager : {proposal['trait'].name} must not be None")
+        if not os.path.isabs(proposal["value"]):
+            raise TraitError(f"FileIdManager : {proposal['trait'].name} must be an absolute path")
+        return proposal["value"]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def index(self, path: str) -> Union[int, str, None]:
```

Comment:
To the best of my knowledge, this is not a public method and should not exist on the ABC.